src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -357,14 +367,29 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
    if (is.null(outputFile))
       outputFile <- .rs.withChangedExtension(inputFile, ext = ".nb.html")
 
-   # TODO: pass encoding from frontend
-   encoding <- "UTF-8"
+   # attempt to get encoding from source database (note: this will only
+   # succeed for files already open in the IDE, but since this operation
+   # is normally called when attempting to preview / create a notebook on
+   # save we generally expect the document to be available)
+   properties <- .rs.getSourceDocumentProperties(inputFile)
+   
+   # attempt to read and re-encode the file to UTF-8 if it's specified
+   # in the system encoding
+   if (!identical(properties$encoding, "UTF-8")) {
+      contents <- readLines(inputFile)
+      utf8 <- tryCatch(
+         iconv(contents, from = properties$encoding, to = "UTF-8"),
+         error = function(e) contents
```

Comment:
If the `iconv` fails, it looks like we'll still write the contents back to the file, and still treat those contents as UTF-8 encoded below. Would it be preferable to just use the encoding as-is in that case?